programs/sll_partition.py

```diff
@@ -0,0 +1,34 @@
+# in a sll make a partition to around a value x. All nodes less than x  come before that all nodes equal or greater to x
+# the x value can appear any where in the right partition. x not needs to appear in the middle, just in right partition.
+
+from single_linked_lists import Node
+
+
+def partition(h: Node, x: int):
+    less_nodes = Node(0)
+    head_less = less_nodes
+    great_nodes = Node(0)
+    head_great = great_nodes
+
+    while h:
+        if h.data < x:
+            less_nodes.next = Node(h.data)
+            less_nodes = less_nodes.next
+        else:
+            great_nodes.next = Node(h.data)
+            great_nodes = great_nodes.next
+
+        if h.next:
+            h = h.next
+        else:
+            break
+    less_nodes.next = head_great.next
```

Comment:
If you use the existing nodes, there are more adjustments that need to be made here.

Reply:
```suggestion
    less_nodes.next = head_great.next
    great_nodes.next = None
```